### PR TITLE
Refactor card scaling into an element for general item scaling

### PR DESCRIPTION
**Pull request by Koshenko**
## About The Pull Request
The current TCG code had some code for scaling its cards down when they're on the ground and then scaling them back in hand/inventory.  This element aims to preserve this functionality and to allow it work for other items.

While the TCG makes the cards smaller on the ground, this element allows for items to be scaled up OR down when on the floor or in inventory.

While this particular element has to do with scaling, I am looking at ways to expand this sort of icon change functionality to `icon_state` as well, but there are additional issues with blood decals needing to be redrawn and possibly `vis_contents`.

This is my first time making an element, so any critiques would be appreciated.

Known issues:
<strike>Currently, cards seems to properly resize when being removed from storage, however there is an issue with the card pile. If you draw a card from the pile and place it back into the pile; then, when performing a click drag on the pile to remove all cards, the drawn card will not scale down (but all the other cards scale properly).  
This only seems to occur with the last card drawn and put back into the pile.  If you don't draw any cards and place them back, then every card scales properly.  This likely has something to do with `handle_empty_deck`.</strike>

Found the issue.  There was a `forceMove` performed on the last card which doesn't send any signal I'm looking for.  Changed it to `remove_from_storage` to get proper signals.

## Why It's Good For The Game
This will not only reduce the amount of code inside of the TCG system, but it will also generalize item scaling to be added to any item.

## Changelog
:cl:
refactor: TCG item scaling refactored into an element
/:cl:

### CODE REVIEW

File: code/datums/elements/item_scaling.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	///Object scaled when placed in an equipment slot OR when entering a storage component.
+	RegisterSignal(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_STORAGE_ENTERED), .proc/scale_storage)
+	///Object scaled when dropped/thrown OR when exiting a storage component.
+	RegisterSignal(target, list(COMSIG_ITEM_DROPPED, COMSIG_STORAGE_EXITED), .proc/scale_overworld)
+
+/datum/element/item_scaling/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED, COMSIG_STORAGE_ENTERED,\
+	COMSIG_STORAGE_EXITED))
+
```
Comment (Rohesie) [inline code/datums/elements/item_scaling.dm:28]:
```suggestion
	UnregisterSignal(target, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED, COMSIG_STORAGE_ENTERED,\
	COMSIG_STORAGE_EXITED))
	return ..()

```
Let's call parent afterwards on unwrapping procs.

File: code/datums/elements/item_scaling.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	src.storage_scaling = storage_scaling
+
+	///Object scaled when placed in an equipment slot OR when entering a storage component.
+	RegisterSignal(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_STORAGE_ENTERED), .proc/scale_storage)
+	///Object scaled when dropped/thrown OR when exiting a storage component.
+	RegisterSignal(target, list(COMSIG_ITEM_DROPPED, COMSIG_STORAGE_EXITED), .proc/scale_overworld)
+
+/datum/element/item_scaling/Detach(datum/target)
+	UnregisterSignal(target, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED, COMSIG_STORAGE_ENTERED,\
+	COMSIG_STORAGE_EXITED))
```
Comment (Rohesie) [inline code/datums/elements/item_scaling.dm:26]:
```suggestion
	UnregisterSignal(target, list(
		COMSIG_ITEM_PICKUP,
		 COMSIG_ITEM_DROPPED,
		 COMSIG_STORAGE_ENTERED,
		COMSIG_STORAGE_EXITED,
	))
```
This works fine, no need to escape the new line.
Trailing commas are good, as well.

File: code/datums/elements/item_scaling.dm [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		return ELEMENT_INCOMPATIBLE
+	///Initial scaling set to overworld_scaling when item is spawned.
+	scale(target, overworld_scaling)
+
+	src.overworld_scaling = overworld_scaling
+	src.storage_scaling = storage_scaling
+
+	///Object scaled when placed in an equipment slot OR when entering a storage component.
+	RegisterSignal(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_STORAGE_ENTERED), .proc/scale_storage)
+	///Object scaled when dropped/thrown OR when exiting a storage component.
```
Comment (Cyberboss) [inline code/datums/elements/item_scaling.dm:21]: Please reserve triple forward slashes for DMDoc comments, not individual lines

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/elements/item_scaling.dm (hunk 3 above)
Comment (Koshenko) [reply code/datums/elements/item_scaling.dm]: I reviewed the DMDoc rules.  I believe my edits should make the documentation up to standards.

**Comment by Rohesie** [comment]
Ready and tested locally?

**Comment by Koshenko** [comment]
> Ready and tested locally?

Yes, and I actually caught a runtime due to some out-of-bounds list accesses when click-dragging a deck to remove all cards.  Latest commit should fix that.  Otherwise, everything is working as intended.

**Pull request merged**